pilot/pkg/xds/delta.go

```diff
@@ -471,15 +471,26 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection, push *model.PushContext,
 		return err
 	}
 
-	// Some types handle logs inside Generate, skip them here.
-	if !gen.Metadata().LogsDetails {
-		if log.DebugEnabled() {
-			// Add additional information to logs when debug mode enabled
-			log.Infof("%s: PUSH for node:%s resources:%d size:%s nonce:%v version:%v",
-				v3.GetShortType(w.TypeUrl), con.proxy.ID, len(res), util.ByteCount(ResourceSize(res)), resp.Nonce, resp.SystemVersionInfo)
+	ptype := "PUSH"
+	info := ""
+	if logdata.Incremental {
+		ptype = "PUSH INC"
+	}
+	if len(info) > 0 {
+		info = " " + logdata.AdditionalInfo
+	}
+
+	if log.DebugEnabled() {
+		// Add additional information to logs when debug mode enabled
+		log.Infof("%s: %s for node:%s resources:%d size:%s nonce:%v version:%v%s",
+			ptype, v3.GetShortType(w.TypeUrl), con.proxy.ID, len(res), util.ByteCount(ResourceSize(res)), resp.Nonce, resp.SystemVersionInfo, info)
+	} else {
+		if logdata.Incremental && log.DebugEnabled() {
+			log.Debugf("%s: %s for node:%s resources:%d size:%s%s",
```

Comment:
this can never be hit, since if `log.DebugEnabled()` is enabled we would always hit the previous branch

Reply:
hmm. Missed the above code. Fixed